app/cafe.py

```diff
@@ -0,0 +1,26 @@
+import datetime
+
+import app.errors as errors
+
+
+class Cafe:
+    def __init__(self, name):
+        self.name = name
+
+    def visit_cafe(self, visitor: dict):

```

Comment:
It's a complicated function to read
You use a lot of nested ifs
You can check requirements one-by-one